clients/src/main/java/org/apache/kafka/clients/admin/DeleteTopicsResult.java

```diff
@@ -30,24 +31,62 @@
  */
 @InterfaceStability.Evolving
 public class DeleteTopicsResult {
-    final Map<String, KafkaFuture<Void>> futures;
+    private Map<String, KafkaFuture<Void>> nameFutures;
+    private Map<Uuid, KafkaFuture<Void>> topicIdFutures;
 
-    protected DeleteTopicsResult(Map<String, KafkaFuture<Void>> futures) {
-        this.futures = futures;
+    protected DeleteTopicsResult() {}
+
+    private void setNameFutures(Map<String, KafkaFuture<Void>> nameFutures) {
+        this.nameFutures = nameFutures;
+    }
+
+    private void setTopicIdFutures(Map<Uuid, KafkaFuture<Void>> topicIdFutures) {
+        this.topicIdFutures = topicIdFutures;
+    }
+
+    protected static DeleteTopicsResult ofTopicNames(Map<String, KafkaFuture<Void>> nameFutures) {
+        DeleteTopicsResult result = new DeleteTopicsResult();
+        result.setNameFutures(nameFutures);
+        return result;
+    }
+
+    protected static DeleteTopicsResult ofTopicIds(Map<Uuid, KafkaFuture<Void>> topicIdFutures) {
+        DeleteTopicsResult result = new DeleteTopicsResult();
+        result.setTopicIdFutures(topicIdFutures);
+        return result;
+    }
+
+    /**
+     * Return a map from topic names to futures which can be used to check the status of
+     * individual deletions if the deleteTopics request used topic names. Otherwise return null.
```

Comment:
nit: can we use javadoc `@return`?

Reply:
I was following the convention from this class before, but I can update.